### PR TITLE
#85 Consider ci-friendly-maven also for archetype

### PR DESCRIPTION
**Pull request by sujith-mn**

### CODE REVIEW

File: templates/server/src/main/resources/META-INF/maven/archetype-metadata.xml
Comment (hohwille) [inline templates/server/src/main/resources/META-INF/maven/archetype-metadata.xml:74]: Can you fix the indendation (here, and in all POMs you changed)... thanks.

File: templates/server/src/main/resources/archetype-resources/pom.xml
Comment (hohwille) [inline templates/server/src/main/resources/archetype-resources/pom.xml:22]: Excellent, this is exactly what we wanted to archive.
Please fix indendation and we can merge.

File: templates/server/src/main/resources/archetype-resources/.mvn/maven.config
Comment (hohwille) [inline templates/server/src/main/resources/archetype-resources/.mvn/maven.config:1]: An archetype can take a version as argument. That is available via `${version}` variable.
It will now be ignored and the initial version is hardcoded to this `0.0.1-SNAPSHOT`.
As `${version}` can also be `1.0.0-SNAPSHOT` we can not easily do
```
-Drevision=${version} -Dchangelist=-SNAPSHOT
```

As this would then result in `1.0.0-SNAPSHOT-SNAPSHOT`.
However, as a nice to have, we could check if via velocity macros it is possible to strip out a potential `-SNAPSHOT` from `${version}`. If that is possible, we should use `${version}` to make it perfect.
However, for me also fine to approve and merge like this.

**Review by hohwille — CHANGES_REQUESTED** [review]
@sujith-mn Thanks for implementing this story. You properly solved this and I wonder why we havent been able to solve this earlier or what have been the problems in the first place. Anyhow it is great that you made it and we finally get this done 👍 

Can you please fix the indendation (see our DoD) so I can approve and merge?
I also added some nice to have imporvement that you can have a look at. But that can be done after merging this PR or never if not easily doable.

File: templates/server/src/main/resources/archetype-resources/.mvn/maven.config
Comment (sujith-mn) [reply templates/server/src/main/resources/archetype-resources/.mvn/maven.config:1]: The formatting is done..

version passing as an argument would be really nice..

Comment (hohwille) [reply templates/server/src/main/resources/archetype-resources/.mvn/maven.config:1]: Thanks for formatting. To prevent blocking this PR, I will merge it now.
Can you please create a new issue for the nice-to-have improvement to put the version here and research velocity macro syntax to remove potential `-SNAPSHOT` suffix?

File: templates/server/src/main/resources/archetype-resources/pom.xml
Comment (hohwille) [inline templates/server/src/main/resources/archetype-resources/pom.xml:33]: Seems Eclipse formatter also indents velocity macros.
We need to check the generated output as this might break the indentation there.
I hoped eclipse would leave things like this as is...
If this is a problem in the resulting pom we can fix in the new story for improvement.

**Review by hohwille — APPROVED** [review]
Thanks. Ready to merge now.

File: templates/server/src/main/resources/archetype-resources/.mvn/maven.config
Comment (sujith-mn) [reply templates/server/src/main/resources/archetype-resources/.mvn/maven.config:1]: Issue #346 created.

is it okay to directly place the version to app.version properties ... like below

#if($version)
	<app.version>${version}</app.version>
#else
	<app.version>${revision}${changelist}</app.version>
#end

**Pull request merged**
